tests/Zend/CodeGenerator/Php/ParameterTest.php

```diff
@@ -134,29 +134,40 @@ public function testFromReflection_GetWithNativeType()
         $reflParam = $this->getFirstReflectionParameter('hasNativeDocTypes');
         $codeGenParam = Zend_CodeGenerator_Php_Parameter::fromReflection($reflParam);
 
-        $this->assertNotEquals('int', $codeGenParam->getType());
-        $this->assertEquals('', $codeGenParam->getType());
+        if (PHP_VERSION_ID < 80000) {
+            $this->assertEquals('', $codeGenParam->getType());
+            $this->assertNotEquals('int', $codeGenParam->getType());
+        } else {
+            $this->assertEquals('int', $codeGenParam->getType());
+        }
     }
 
-    static public function dataFromReflection_Generate()
+    public static function dataFromReflection_Generate()
     {
-        return array(
-            array('name', '$param'),
-            array('type', 'stdClass $bar'),
-            array('reference', '&$baz'),
-            array('defaultValue', '$value = \'foo\''),
-            array('defaultNull', '$value = null'),
-            array('fromArray', 'array $array'),
-            array('hasNativeDocTypes', '$integer'),
-            array('defaultArray', '$array = array ()'),
-            array('defaultArrayWithValues', '$array = array (  0 => 1,  1 => 2,  2 => 3,)'),
-            array('defaultFalse', '$val = false'),
-            array('defaultTrue', '$val = true'),
-            array('defaultZero', '$number = 0'),
-            array('defaultNumber', '$number = 1234'),
-            array('defaultFloat', '$float = 1.34'),
-            array('defaultConstant', '$con = \'foo\'')
-        );
+        $commonValues =  array(
+        array('name', '$param'),
+        array('type', 'stdClass $bar'),
+        array('reference', '&$baz'),
+        array('defaultValue', '$value = \'foo\''),
+        array('defaultNull', '$value = null'),
+        array('fromArray', 'array $array'),
+        array('defaultArray', '$array = array ()'),
+        array('defaultArrayWithValues', '$array = array (  0 => 1,  1 => 2,  2 => 3,)'),
+        array('defaultFalse', '$val = false'),
+        array('defaultTrue', '$val = true'),
+        array('defaultZero', '$number = 0'),
+        array('defaultNumber', '$number = 1234'),
+        array('defaultFloat', '$float = 1.34'),
+        array('defaultConstant', '$con = \'foo\'')
+    );
+
+        if (PHP_VERSION_ID < 80000) {
+            $commonValues[] = array('hasNativeDocTypes', '$integer');
+        } else {
+            $commonValues[] = array('hasNativeDocTypes', 'int $integer');
+        }
+
+        return $commonValues;
```

Comment:
may I suggest this instead:
```suggestion
        return array(
            array('name', '$param'),
            array('type', 'stdClass $bar'),
            array('reference', '&$baz'),
            array('defaultValue', '$value = \'foo\''),
            array('defaultNull', '$value = null'),
            array('fromArray', 'array $array'),
            array('hasNativeDocTypes', PHP_VERSION_ID >= 80000 ? 'int $integer' : '$integer'),
            array('defaultArray', '$array = array ()'),
            array('defaultArrayWithValues', '$array = array (  0 => 1,  1 => 2,  2 => 3,)'),
            array('defaultFalse', '$val = false'),
            array('defaultTrue', '$val = true'),
            array('defaultZero', '$number = 0'),
            array('defaultNumber', '$number = 1234'),
            array('defaultFloat', '$float = 1.34'),
            array('defaultConstant', '$con = \'foo\'')
        );
```

Reply:
Fine by me